com.unity.shadergraph/Editor/Data/Graphs/VirtualTextureShaderProperty.cs

```diff
@@ -75,7 +75,11 @@ internal override void ForeachHLSLProperty(Action<HLSLProperty> action)
             int numLayers = value.layers.Count;
             if (numLayers > 0)
             {
-                action(new HLSLProperty(HLSLType._CUSTOM, referenceName + "_CBDecl", HLSLDeclaration.UnityPerMaterial, concretePrecision)
+                HLSLDeclaration decl = HLSLDeclaration.UnityPerMaterial;
+                if (value.procedural)
+                    decl = GetDefaultHLSLDeclaration();
```

Comment:
Procedural VT is always set up as Global at the moment, so this will always return HLSLDeclaration.Global